### PR TITLE
Add rake task to run spec for all Ruby versions

### PR DESCRIPTION
**Pull request by lokeshh**
For #124 

Create a rake task to test the specs in all Ruby versions automatically and display the result.

Tests for all versions are ran using `rake all`.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -2,6 +2,16 @@ require 'rspec/core/rake_task'
 require 'bundler/gem_tasks'
 
 lib_folder = File.expand_path("../lib", __FILE__)
+
+RUBIES = ['ruby-2.0', 'ruby-2.1', 'ruby-2.2', 'ruby-2.3.0']
+
+task :all do |task|
+  RUBIES.each do |ruby_v|
+    puts "\n\n\nUsing #{ruby_v}\n\n\n"
+    puts `bash -lc "rvm use #{ruby_v}; rspec spec"`
```
Comment (v0dro) [inline Rakefile]: rspec provides a class for defining rake tasks. I think RVM has it's own Ruby API too. Use that. What if the user is not using bash? Or if he does not have bash?

Comment (lokeshh) [inline Rakefile]: Point. I will have a look what RVM API has to offer.

Comment (lokeshh) [inline Rakefile]: Checkout the new solution. It makes use of RVM generated wrappers.

**Comment by v0dro** [comment]
OK so this is one part of the solution. But if there is a failure in Ruby 2.0.0, I will need to scroll all the way up to see that. It should ideally print the test report (number of tests passed and failures) for each ruby version at the bottom of the terminal output, so that in case there are failures, I know where to look instantly.

**Comment by v0dro** [comment]
Btw it should also run rubocop.

**Comment by lokeshh** [comment]
How about just printing the number of passed/failed for each Ruby? The user can just change Ruby version and run rspec if he wants to know about the failures.

**Comment by v0dro** [comment]
Yeah that works too. Make sure you use appropriate colours (red for fail, green for pass).

**Comment by lokeshh** [comment]
Ok. I did like I said but adding colors would need a little more work.

Should the rubocop also run in this same task or should I create another task for it?

**Comment by v0dro** [comment]
You can add the :cop task as a subtask of :all.

**Comment by lokeshh** [comment]
`:cop` doesn't need to run for all rubies right?

**Comment by lokeshh** [comment]
Color support is added. Now `:all` firstly runs `rubocop` then it writes the summary for all rubies.

File: formatter.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def dump_summary(message)
+    colorizer = ::RSpec::Core::Formatters::ConsoleCodes
+
+    output.puts "\nFinished in #{message.formatted_duration} " \
+       "(files took #{message.formatted_load_time} to load)\n" \
+       "#{message.colorized_totals_line(colorizer)}\n"
+  end
+end
```
Comment (lokeshh) [inline formatter.rb]: Thanks to [patkoperwas](http://stackoverflow.com/users/3295284/patkoperwas) for he made it possible just print the number of pass/failed examples with colors. [Here's](http://stackoverflow.com/questions/37553229/print-only-the-number-of-failed-examples-with-rspec?noredirect=1#comment62596136_37553229) his answer that helped me.

Comment (v0dro) [inline formatter.rb]: How about rename this file to `.rspec_formatter.rb` so that its explicit that it's rspec and it stays hidden too.

Comment (lokeshh) [inline formatter.rb]: Done

**Comment by lokeshh** [comment]
@v0dro I think this is ready to be merged. Just a small thing. Please specify the complete versions names to be tested. Currently I have these: https://github.com/v0dro/daru/pull/129/files#diff-52c976fc38ed2b4e3b1192f8a8e24cffR7

**Pull request merged**